### PR TITLE
Update README for Discovery SPI impl

### PR DESCRIPTION
**Pull request by emrahkocaman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,106 @@
...
-To configure your cluster to use EC2 auto-discovery, follow the below steps:
+There are two possible ways to configure your cluster to use EC2 auto-discovery.
+You can either choose to configure your cluster with `AwsConfig` (or `aws` element in your XML config)
+or you can choose configuring AWS discovery using [Discovery SPI](http://docs.hazelcast.org/docs/latest/manual/html-single/index.html#discovery-spi) implementation.
+Starting with hazelcast-aws-1.2, it's strongly recommended to use Discovery SPI implementation as
+the former one will be deprecated.
+
+###Configuring AWS Discovery using Discovery SPI for Hazelcast Cluster Members
+
+- Add the *hazelcast-aws.jar* dependency to your project. Note that it is also bundled inside *hazelcast-all.jar*. The Hazelcast aws module does not depend on any other third party modules.
```
Comment (Serdaro) [inline README.md]: I think module name is used as "hazelcast-aws" throughout the README. So for the sake of consistency "The Hazelcast aws module ...." can be changed as "The hazelcast-aws module..."

File: README.md [hunk 2]
```diff
@@ -27,7 +27,106 @@
...
+- Disable join over multicast, TCP/IP and aws by setting the `enabled` attribute of the related tags to `false`.
```
Comment (Serdaro) [inline README.md]: ....TCP/IP and AWS by setting....

File: README.md [hunk 3]
```diff
@@ -27,7 +27,106 @@
...
+                   <property name="hz-port">5701</property>
+                </properties>
+            </discovery-strategy>
+        </discovery-strategies>
+    </join>
+  </network>
+ </hazelcast>
+```
+
+Here are the definitions of properties
```
Comment (Serdaro) [inline README.md]: ...of the properties:

File: README.md [hunk 4]
```diff
@@ -27,7 +27,106 @@
...
+
+* `access-key`, `secret-key`: Access and secret keys of your account on EC2.
+* `iam-role`: If you do not want to use access key and secret key, you can specify `iam-role`. Hazelcast-aws plugin fetches your credentials by using your IAM role. It is optional.
```
Comment (Serdaro) [inline README.md]: Is it plugin or module? Again for the sake of consistency, let's choose one and use it for all hazelcast-aws phrases.

File: README.md [hunk 5]
```diff
@@ -27,7 +27,106 @@
...
+* `tag-key`, `tag-value`: To narrow the members in the cloud down to only Hazelcast members, you can set these parameters as the ones you specified in the EC2 console. They are optional.
+* `connection-timeout-seconds`: The maximum amount of time Hazelcast will try to connect to a well known member before giving up. Setting this value too low could mean that a member is not able to connect to a cluster. Setting the value too high means that member startup could slow down because of longer timeouts (for example, when a well known member is not up). Increasing this value is recommended if you have many IPs listed and the members cannot properly build up the cluster. Its default value is 5.
+* `hz-port`: You can set searching for other ports rather than 5701 if you've members on different ports. It is optional.
+
+###Configuring AWS Discovery using Discovery SPI for Hazelcast Client
+
+- Add the *hazelcast-aws.jar* dependency to your project. Note that it is also bundled inside *hazelcast-all.jar*. The Hazelcast aws module does not depend on any other third party modules.
```
Comment (Serdaro) [inline README.md]: ....inside *hazelcast-all.jar*. Hazelcast-aws module (or plugin?) does not....

File: README.md [hunk 6]
```diff
@@ -27,7 +27,106 @@
...
+            </properties>
+        </discovery-strategy>
+    </discovery-strategies>
+  </network>
+ </hazelcast-client>
+```
+
+
+
+List of available properties and their documentation can be found at [AwsProperties](https://github.com/hazelcast/hazelcast-aws/blob/master/src/main/java/com/hazelcast/aws/AwsProperties.java)
```
Comment (Serdaro) [inline README.md]: ..../AwsProperties.java).

File: README.md (hunk 5 above)
Comment (emrahkocaman) [inline README.md]: Good catch, this is an outdated statement. 👍

**Comment by emrahkocaman** [comment]
@Serdaro Thanks for the review,  I've made the changes you requested. Can you please give another look?

**Comment by Serdaro** [comment]
Ok from me.

**Pull request merged**